tests/Unit/NumericalAlgorithms/Spectral/Test_SwshCollocation.cpp

```diff
@@ -0,0 +1,130 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cmath>
+#include <cstddef>
+#include <random>
+#include <sharp_cxx.h>
+
+#include "NumericalAlgorithms/Spectral/ComplexDataView.hpp"
+#include "NumericalAlgorithms/Spectral/SwshCollocation.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+
+namespace Spectral {
+namespace Swsh {
+namespace {
+
+// IWYU pragma: no_include <sharp_geomhelpers.h>
+// IWYU pragma: no_include <sharp_lowlevel.h>
+
+template <ComplexRepresentation Representation>
+void test_spherical_harmonic_collocation() noexcept {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<size_t> sdist{8, 64};
+  size_t l_max = sdist(gen);
+
+  CAPTURE(l_max);
+  Collocation<Representation>& precomputed_collocation =
+      precomputed_spherical_harmonic_collocation<Representation>(l_max);
+
```

Comment:
Since `precomputed_collocation` is non-const, you should mutate it here, just to hit your caching bug :) (though this will be fixed by returning a `const &` from precomputed_spherical_harmonic_collocation)